pointers/pointers.c

```diff
@@ -21,52 +23,97 @@ void swap(int* a, int* b)
 */
 char *find_char(char *str, int c)
 {
-
+  char *i = str;
+  while(*i != '\0'){
+    if(*i == c){
+      return i;
+    }
+    i++;
+  }
+  return NULL;
 }
 
 /*
     Given an empty (NULL) character pointer x and a character pointer y,
     copies the character contents of y over to x. Again, pointer arithmetic
-    is necessary here. Also, make sure x points to a null character at its 
-    end to terminate it properly. 
-    
+    is necessary here. Also, make sure x points to a null character at its
+    end to terminate it properly.
+
     Do not just use the `strcpy` function from the standard library.
 */
 void string_copy(char *x, char *y)
 {
-
+  while(*y != '\0'){
+    *x = *y;
+    y++;
+    x++;
+  }
+  *x = '\0';
```

Comment:
Great implementation here! Seems like you have a great understanding of pointers already.